### ISSUE TITLE
Kestrel HTTPS + WebSockets + Safari fails on 1.1.0

### ISSUE THREAD

**Issue by natemcmaster** [aspnet/KestrelHttpServer#1212]
**Repro**
Run the EchoApp from aspnet/WebSockets (https://github.com/aspnet/WebSockets/tree/rel/1.1.0/samples/EchoApp) from any OS (windows, linux, mac).
Add https
Open app in Safari
Try to open new web socket

**Result**
Page opens fine (https)
Web sockets fail to open (wss)

**Error**
In browser: 
![image](https://cloud.githubusercontent.com/assets/2696087/20281821/db2edc3e-aa66-11e6-9e85-1fe222552c22.png)

On server:
```
info: HttpsConnectionFilter[1]
      Failed to authenticate HTTPS connection.
System.IO.IOException: Authentication failed because the remote party has closed the transport stream.
   at System.Net.Security.SslState.StartReadFrame(Byte[] buffer, Int32 readBytes, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslState.PartialFrameCallback(AsyncProtocolRequest asyncRequest)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Net.Security.SslState.InternalEndProcessAuthentication(LazyAsyncResult lazyResult)
   at System.Net.Security.SslState.EndProcessAuthentication(IAsyncResult result)
   at System.Net.Security.SslStream.EndAuthenticateAsServer(IAsyncResult asyncResult)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Server.Kestrel.Https.HttpsConnectionFilter.<OnConnectionAsync>d__6.MoveNext()
```

**Info**
CLI: 1.0.0-preview2-1-003177
.NET Core: 1.1.0
.Kestrel.Https: 1.1.0
Server OS: Any operating system
Browser: Safari 10, macOS 10.12

**Remarks**
Doesn't repro in Chrome on macOS

**Comment by natemcmaster** [comment]
cc @anurse

**Comment by natemcmaster** [comment]
Nevermind, this is a client issue. In Safari, web sockets fail when the certificate is bad. The testCert.pfx I was using had expired. Even if you "always trust localhost", because it is expired, Safari still rejects the cert and won't open the connection.

<img width="400" alt="screen shot 2016-11-14 at 2 39 52 pm" src="https://cloud.githubusercontent.com/assets/2696087/20285720/3fce7efe-aa78-11e6-8920-57c707de1151.png">

That said, we can probably harden Kestrel to handle the IOException that comes from this scenario.

**Comment by halter73** [comment]
The logging issue seems similar to #960. The fix in #1017 was to catch the exception in the HttpsConnectionFilter and lower the log level to info instead of error. I think it's reasonable to log the error with the IOException thrown from SSL stream at the info level. Certainly it should be logged at some level.

I would close this.
